nixos/modules/services/misc/nix-daemon.nix

```diff
@@ -734,7 +734,7 @@ in
             CPUSchedulingPolicy = cfg.daemonCPUSchedPolicy;
             IOSchedulingClass = cfg.daemonIOSchedClass;
             IOSchedulingPriority = cfg.daemonIOSchedPriority;
-            LimitNOFILE = 4096;
+            LimitNOFILE = "infinity";
```

Comment:
I'd prefer a very high number, so that we don't accidentally DoS the whole system.
These are in-kernel data structures after all.

Reply:
Seems like a good pick @Artturin 

![one million fds](https://user-images.githubusercontent.com/496447/172258197-9e2670a1-5d04-4886-93c0-0ca4ec2dfe19.gif)

My system is at 50K right now, so this is probably in the ok ballpark. We can wiggle it around if we learn more.